privacy_evaluator/attacks/property_inference_attack.py

```diff
@@ -1,33 +1,115 @@
+from ..attacks.attack import Attack
+from ..classifiers.classifier import Classifier
+from ..utils import data_utils
+from ..utils.trainer import trainer
+from ..models.tf.conv_net_meta_classifier import ConvNetMetaClassifier
+from ..utils.model_utils import copy_and_reset_model
+
 
 import numpy as np
 import torch
+from torch import nn
 import tensorflow as tf
-from sklearn.model_selection import train_test_split
-from typing import Tuple, Dict, List
-from art.estimators.classification import TensorFlowV2Classifier
-import string
+from tqdm import tqdm
+import sys
+from typing import Tuple, Dict, List, Union
+from art.estimators.classification import TensorFlowV2Classifier, PyTorchClassifier
+from collections import OrderedDict
+import warnings
+
+# count of shadow training sets, must be even
+AMOUNT_SETS = 2
+# ratio and size for unbalanced data sets
+SIZE_SET = 1000
+# ratios for different properties in sub-attacks
+RATIOS_FOR_ATTACK = [
+    0.05,
+    0.1,
+    0.15,
+    0.2,
+    0.25,
+    0.3,
+    0.35,
+    0.4,
+    0.45,
+    0.55,
+    0.6,
+    0.65,
+    0.7,
+    0.75,
+    0.8,
+    0.85,
+    0.9,
+    0.95,
+]
+# classes the attack should be performed on
+CLASSES = [0, 1]
 
 
 class PropertyInferenceAttack(Attack):
     def __init__(
-        self, target_model: Classifier, dataset: Tuple[np.ndarray, np.ndarray]
+        self,
+        target_model: Classifier,
+        dataset: Tuple[np.ndarray, np.ndarray],
+        amount_sets: int = AMOUNT_SETS,
+        size_set: int = SIZE_SET,
+        ratios_for_attack: List[int] = RATIOS_FOR_ATTACK,
+        classes: List[int] = CLASSES,
+        verbose: int = 0,
     ):
         """
         Initialize the Property Inference Attack Class.
         :param target_model: the target model to be attacked
         :param dataset: dataset for training of shadow classifiers, test_data from dataset
+        :param amount_sets: count of shadow training sets, must be even
+        :param size_set: ratio and size for unbalanced data sets
+        :param ratios_for_attack: ratios for different properties in sub-attacks
         with concatenation [test_features, test_labels]
+        :param classes: classes the attack should be performed on
+        :param verbose: 0: no information; 1: backbone (most important) information; 2: utterly detailed information will be printed
         """
+        if not (
+            isinstance(dataset, tuple)
+            and list(map(type, dataset)) == [np.ndarray, np.ndarray]
+        ):
+            raise TypeError("Dataset type should be of shape (np.ndarray, np.ndarray).")
+
         self.dataset = dataset
-        # count of shadow training sets, must be eval
-        self.amount_sets = 2
+        if not (
+            isinstance(target_model, TensorFlowV2Classifier)
+            or isinstance(target_model, PyTorchClassifier)
+        ):
+            raise TypeError("Target model must be of type Classifier.")
```

Comment:
You could be more precise here. Might be that someone inserts TensorFlow 1 classifiers. This user will be confused. So say very explicitly which two types you accept.